File: 250714/숫자 빠르게 찾기/find-number-fast.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const [n, m] = input[0].split(' ').map(Number);
const numbers = input[1].split(' ').map(Number);
const queries = input.slice(2, 2 + m).map(Number);

function binarySearch(arr, target){
    let left = 0;
    let right = arr.length - 1;
    while(left <= right){
        let mid = Math.floor((left + right)/2);
        if(arr[mid] === target){
            return mid + 1
        }
        if(arr[mid] > target){
            right = mid - 1
        }
        if(arr[mid] < target){
            left = mid + 1
        }
    }
    return -1;
}

queries.forEach(val => {
    const a = binarySearch(numbers, val)
    console.log(a)
})

File: 250918/숫자 빠르게 찾기/find-number-fast.py
n, m = map(int, input().split())
arr = list(map(int, input().split()))
queries = [int(input()) for _ in range(m)]

def binary(val):
    left, right = 0, n-1

    while left <= right:
        mid = (left + right) // 2
        if arr[mid] == val:
            return mid
        if arr[mid] > val:
            right = mid - 1
        else:
            left = mid + 1
    
    return -1

for val in queries:
    idx = binary(val)
    print(-1 if idx == -1 else idx + 1)